bot/extensions/clashofcode/commands.py

```diff
@@ -0,0 +1,290 @@
+                "You can't create a clash unless you participate in the session\n"
+                "Use `/coc session join` or react to the pinned message to join the coc session "
+                "<:smilecat:727592135171244103>",
+                ephemeral=True,
+            )
+
+        link = REGEX.fullmatch(url)
+        if not link:
+            return await interaction.response.send_message('Could not find any valid "clashofcode" url', ephemeral=True)
+
+        # Defer the response to acknowledge the interaction while doing slow tasks
+        await interaction.response.defer(ephemeral=True, thinking=True)
+
+        self.last_clash = time.time()
+        session_id = link[1]
+
+        async with aiohttp.ClientSession() as session:
+            async with session.post(API_URL, json=[session_id]) as resp:
+                json = await resp.json()
+
+        plang = json["programmingLanguages"]
+        pager = commands.Paginator(
+            prefix="\n".join(
+                [
+                    f"**Hey, {interaction.user.mention} is hosting a Clash Of Code game!**",
+                    f"Mode{'s' if len(json['modes']) > 1 else ''}: {', '.join(json['modes'])}",
+                    f"Programming languages: {', '.join(plang) if plang else 'All'}",
+                    f"Join here: {link[0]}",
+                ]
+            ),
+            suffix="",
+        )
+
+        for member_id in coc_helper.session_users:
+            if member_id != interaction.user.id:
+                member = self.bot.get_user(member_id)
+                pager.add_line(member.mention + ", ")
+
+        if not len(pager.pages):
+            return await interaction.followup.send(
+                "Nobody is online to play with <:pepesad:733816214010331197>", ephemeral=True
+            )
+
+        for page in pager.pages:
+            await interaction.channel.send(page, allowed_mentions=discord.AllowedMentions(users=True))
+
+        await interaction.delete_original_response()
+
+        async with aiohttp.ClientSession() as session:
+            while not json["started"]:
+                await asyncio.sleep(10)  # wait 10s to avoid flooding the API
+                async with session.post(API_URL, json=[session_id]) as resp:
+                    json = await resp.json()
+
+        players = len(json["players"])
+        players_text = ", ".join(
+            [p["codingamerNickname"] for p in sorted(json["players"], key=lambda p: p["position"])]
+        )
+        start_message = await interaction.channel.send(embed=em(json["mode"], players_text))
+
+        async with aiohttp.ClientSession() as session:
+            while not json["finished"]:
+                await asyncio.sleep(10)  # wait 10s to avoid flooding the API
+                async with session.post(API_URL, json=[session_id]) as resp:
+                    json = await resp.json()
+
+                if len(json["players"]) != players:
```

Comment:
I was probably the one who wrote this but this condition is only triggered if a players leaves I think, it would be better to check whether the `rank` of the players have changed since the last API call. 

NB: the `position` is the order in which players joined and the `rank` is the current order of the solutions, if players haven't submitted yet their `rank` are all last so the ordering should be on `rank` first then on `position` if the `rank` are equal.